pages/api/efd/[type].js

```diff
@@ -29,12 +29,10 @@ export default function handler(req, res) {
       data.push(tableMeta.toObject(row));
     },
     error(error) {
-      // reject(error);
-      res.status(200).json({ error })
+      return res.status(405).json(error);
     },
     complete() {
-      // resolve(data);
-      res.status(200).json({ data })
+      return res.status(200).json(type === "current" ? data[0] : data);
```

Comment:
I was doing 2 dumb things:

1. I was returning objects that looked like `{ data: { data: [stuff, and, things] } }` when it shoulda been `{ data: [stuff, and, things] }`
2. everywhere `currentData` was being used expected a single result instead of an array of results, so I adjusted here instead of downstream.  That is, I suppose a different way to approach could be to have all `summitData` consumers all always expect an array of data.  Up to @alexgoff which approach is best fit.